### PR TITLE
Implemented basic multisearch endpoint support

### PR DESCRIPTION
**Pull request by javiertoledo**
Hi @jaeyson,

I added this implementation to enable the `multi_search` endpoint for vector searches. When using vector embeddings, the request size tends to be much larger, and the Typesense documentation recommends using the multi-search endpoint to avoid potential limits using GET requests.

I've implemented a small test for the vector embedding scenario and drafted the @doc comments, trying to mimic your code style. However, I haven't implemented an integration with Ecto because I didn't need it. Please let me know your thoughts, and feel free to close the PR and rework it in your way.

Thanks again for this nice library!

### CODE REVIEW

**Comment by jaeyson** [comment]
Hi @javiertoledo thank you for your contrib! 🙏🏾

There's one change that we need for this (based on the recent PR #30): We changed the connection struct into a plain map, so long as it matches the [type](https://hexdocs.pm/ex_typesense/ExTypesense.Connection.html#types)

```elixir
@type t() :: %{
  host: String.t() | nil,
  api_key: String.t() | nil,
  port: non_neg_integer() | nil,
  scheme: String.t() | nil
}
```

File: test/multi_search_test.exs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+  setup_all do
+    conn = %ExTypesense.Connection{
+      host: "localhost",
+      api_key: "xyz",
+      port: 8108,
+      scheme: "http"
+    }
```
Comment (jaeyson) [inline test/multi_search_test.exs:14]: you can remove this connection struct, as I've already added the connection string in `test/test_helper.exs`

File: test/multi_search_test.exs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    document = %{
+      collection_name: "companies",
+      company_name: "Test",
+      company_id: 1001,
+      country: "US",
+      company_description: "This is a test company",
+      company_description_embedding: @embedding
+    }
+
+    ExTypesense.create_collection(conn, schema)
```
Comment (jaeyson) [inline test/multi_search_test.exs:37]: No need to pass `conn`, just `create_collection(schema)` is fine.

**Comment by jaeyson** [comment]
Hi @javiertoledo I can merge this to main, with the changes needed that I commented.

Here's what I'll do:

- merge this so that you can use it right away (assuming changes we're done).
- I'll add the Ecto part and with the updates need.

just so there's no blocker on using the library. How's that working for you?

**Comment by javiertoledo** [comment]
Sure thing, @jaeyson. I'm using this change from my fork and it's working fine for me, but feel free to pull the branch and change it on your side. Once everything is merged and a new version released, I'll redirect my mix.exs back to the official package.

**Comment by jaeyson** [comment]
Hi @javiertoledo, I'll transfer this PR to #35 (kindly check). Again, thank you for helping me out!

**Pull request closed without merging**